### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by ifranzki**

### CODE REVIEW

File: usr/lib/ep11_stdll/ep11_specific.c [hunk 1]
```diff
@@ -1550,6 +1550,16 @@ static CK_RV ab_unwrap_update_template(STDLL_TokData_t * tokdata,
         return CKR_TEMPLATE_INCONSISTENT;
     }
     for (i = 0; i < sizeof(attrs) / sizeof(CK_ATTRIBUTE); ++i) {
+        if (attrs[i].type == CKA_VALUE_LEN) {
+            switch (keytype) {
+            case CKK_GENERIC_SECRET:
```
Comment (jschmidb) [inline usr/lib/ep11_stdll/ep11_specific.c:1555]: What's about CKK_DES2 and CKK_DES3? They show up in ep11_supported_mech_list()

File: usr/lib/ep11_stdll/ep11_specific.c [hunk 2]
```diff
@@ -8129,6 +8129,13 @@ CK_RV ep11tok_generate_key_pair(STDLL_TokData_t * tokdata, SESSION * sess,
...
     }
 
+    rc = update_ep11_attrs_from_blob(tokdata, sess, public_key_obj, FALSE);
+    if (rc != CKR_OK) {
+        TRACE_ERROR("%s update_ep11_attrs_from_blob failed with rc=0x%lx\n",
+                    __func__, rc);
+        goto error;
+    }
+
     /* Copy CKA_MODULUS and CKA_PUBLIC_EXPONENT attributes from
```
Comment (jschmidb) [inline usr/lib/ep11_stdll/ep11_specific.c:8139]: ok, yes, that was missing. Not all attrs handled by update_ep11_attrs_from_blob() may be applicable for public keys.

Comment (jschmidb) [reply usr/lib/ep11_stdll/ep11_specific.c:8139]: ... but this is handled by checking each attr value for CK_UNAVAILABLE_INFORMATION. OK.

File: usr/lib/ep11_stdll/ep11_specific.c (hunk 1 above)
Comment (ifranzki) [reply usr/lib/ep11_stdll/ep11_specific.c:1555]: DES keys are fixed in length, and the length only depends on the key type (DES, DES2 or DES3). Thus, the PKCS#11 standard does not define the CKA_VALUE_LEN attribute for such keys.  The only key types where CKA_VALUE_LEN is defined per PKCS#11 standard is AES (i.e. 128, 192, 256) and generic secret.

**Review by jschmidb — APPROVED** [review]

**Pull request merged**